sklearn/neighbors/lof.py

```diff
@@ -0,0 +1,288 @@
+# -*- coding: utf-8 -*-
+
+import numpy as np
+
+from .base import NeighborsBase
+from .base import KNeighborsMixin
+from .base import UnsupervisedMixin
+
+from ..metrics import pairwise_distances
+
+from ..utils import check_array
+from ..utils import _get_n_jobs
+
+__all__ = ["LOF"]
+
+
+
+class LOF(NeighborsBase, KNeighborsMixin, UnsupervisedMixin):
+    """Unsupervised Outlier Detection.
```

Comment:
Local Outlier Factor (LOF)

LOF is an unsupervised outlier detection method. It returns ...